tutorials/composition/Interpolated Waveforms.ipynb

```diff
@@ -0,0 +1,314 @@
+{
```

Comment:
<p>I would maybe add a small part (such as this one where 'kind' is explained) to tell about pcw constant interpolation which is used in QAOA or others optimal control methods. You can do it using ConstantPulse and add them one by one to a sequence but here you can just provide times and values. </p><p>Don't know if it's worth it, just an idea</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pasqal-io/Pulser/pull/224/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Reply:
<p>I think you would do this better than me, do you want to add it yourself?</p>

Reply:
<p>Ok, I'll try something and you'll choose if it can be added or not</p>

Reply:
<p>Thanks a lot!</p>

Reply:
<p>It would look like this </p><p> </p><p>layer = 3</p><p>values = np.array([1,0] * layer + [0])</p><p>ts = np.sort([0]+np.random.uniform(0,1,2*layer-1).tolist()+[1])</p><p>int_wf4 = InterpolatedWaveform(duration, values, times = ts, interpolator = "interp1d", kind = "previous")</p><p>int_wf4.draw()</p><p> </p><p>Actually, after writing it, it seems a bit out of context here. And it might overlap with the  "add constant pulses in sequence with loop" in the QAOA parametrized tutorial.</p><pre class="ql-syntax" spellcheck="false">for t, s in zip(t_list, s_list): 
    pulse_1 = Pulse.ConstantPulse(1000*t, 1., 0., 0) 
    pulse_2 = Pulse.ConstantPulse(1000*s, 1., 1., 0)

    seq.add(pulse_1, 'ch0')
    seq.add(pulse_2, 'ch0')
</pre><p> </p><p>So maybe it's best not to add it. </p>

Reply:
<p>So you'd suggest we had it to QAOA tutorials instead?</p>